### PR TITLE
Feature/sg 34 regseg transfer learning

### PR DESCRIPTION
**Pull request by shaydeci**
Added example code for Regseg transfer learning, which will later be used for an example notebook.
This PR holds several necessary features for it to run:

- BCE with Dice loss.
- Binary IOU metric object (I.e IOU only for target class).
- Binary segmentation visualisation callback.
- Supervisely dataset interface.
- Different lr assignment for head and backbone for RegSeg.

### CODE REVIEW

File: src/super_gradients/training/metrics/segmentation_metrics.py [hunk 1]
```diff
@@ -111,3 +111,16 @@ def update(self, preds, target: torch.Tensor):
             preds = preds[0]
         _, preds = torch.max(preds, 1)
         super().update(preds=preds, target=target)
+
+
+class BinaryIOU(torchmetrics.IoU):
```
Comment (lotem-deci) [inline src/super_gradients/training/metrics/segmentation_metrics.py:116]: @shaydeci  - Did we make a decision on what measures we are going to take to handle the SegmentationMetrics issue we've had with mIOU (caused by ignore_index...)?

File: src/super_gradients/training/utils/callbacks.py [hunk 2]
```diff
@@ -477,6 +478,30 @@ def __call__(self, context: PhaseContext):
             context.sg_logger.add_images(tag=tag, images=batch_imgs[:self.last_img_idx_in_batch], global_step=context.epoch, data_format='NHWC')
 
 
+class BinarySegmentationVisualizationCallback(PhaseCallback):
```
Comment (lotem-deci) [inline src/super_gradients/training/utils/callbacks.py:481]: @shaydeci  - Why did we implement a specific VisualizationCallback for BinaryIOU? Can't we have a generic implementation for SegmentationIOU and have BinaryIOU as a specific edge case of it?

Comment (lotem-deci) [reply src/super_gradients/training/utils/callbacks.py:481]: If it's complicated and will slow us down let's open an issue in the backlog for it - But I am not sure it cannot be implemented easily for more than 2 classes

File: src/super_gradients/training/utils/segmentation_utils.py [hunk 3]
```diff
@@ -350,3 +373,100 @@ def to_one_hot(target: torch.Tensor, num_classes: int, ignore_index: int = None)
         one_hot = torch.cat([one_hot[:, :ignore_index], one_hot[:, ignore_index + 1:]], dim=1)
 
     return one_hot
+
+
+def undo_image_preprocessing(im_tensor: torch.Tensor) -> np.ndarray:
```
Comment (lotem-deci) [inline src/super_gradients/training/utils/segmentation_utils.py:378]: @shaydeci  - Can we give this function a better name?

File: src/super_gradients/training/utils/segmentation_utils.py [hunk 4]
```diff
@@ -21,6 +26,19 @@ def __repr__(self):
         return self.__class__.__name__ + str(self.__dict__).replace('{', '(').replace('}', ')')
 
 
+class ResizeSeg(SegmentationTransform):
```
Comment (lotem-deci) [inline src/super_gradients/training/utils/segmentation_utils.py:29]: @shaydeci  - We need to create a proper place in the modules hierarchy for all of the Augmentations instead of putting everything inside utils...

**Comment by lotem-deci** [comment]
Nice work.
We have some required fixes inline...

File: src/super_gradients/training/metrics/segmentation_metrics.py (hunk 1 above)
Comment (shaydeci) [reply src/super_gradients/training/metrics/segmentation_metrics.py:116]: Yes as we discussed- I checked and IOU with ignored index behaviour is as right.
For binary segmentation though it is more common to just take the target IOU, but I'm also returning mIOU and BG IOU for logging.

File: src/super_gradients/training/utils/callbacks.py (hunk 2 above)
Comment (shaydeci) [reply src/super_gradients/training/utils/callbacks.py:481]: Regarding the first question- IOU metric returns the mean across all classe including background, where for binary segmentation though it is more common to just take the target IOU, but I'm also returning mIOU and BG IOU for logging.

Regarding the second question- you are probably right, it just requires some thought about how to display the images (see the way I present TPs, FN, and FPs). Not sure wed want to do that for all classes....maybe we do, but anyways, Il add it to the backlog since I think it will slow us down.

**Review by lotem-deci — APPROVED** [review]
LGTM
